documentation/src/main/asciidoc/public/reference/_indexing-explicit.adoc

```diff
@@ -0,0 +1,281 @@
+while for the <<mapper-pojo-standalone,Standalone POJO Mapper>> the `SearchSession` is <<entrypoints-search-session-mapper-pojo-standalone,is created explicitly by the application>>.
+2. Eventually, the application decides changes are complete,
+and the plan processes change events added so far,
+either inferring which entities need to be reindexed and building the corresponding documents (<<coordination-none,no coordination>>)
+or building events to be sent to the outbox (<<coordination-outbox-polling,`outbox-polling` coordination>>).
++
+The application may trigger this explicitly using the indexing plan's `process` method,
+but it is generally not necessary as it happens automatically:
+for the <<mapper-orm,Hibernate ORM integration>> this happens when the Hibernate ORM `Session` gets flushed
+(explicitly or as part of a transaction commit),
+while for the <<mapper-pojo-standalone,Standalone POJO Mapper>> this happens when the `SearchSession` is closed.
+3. Finally the plan gets executed, triggering indexing, potentially asynchronously.
++
+The application may trigger this explicitly using the indexing plan's `execute` method,
+but it is generally not necessary as it happens automatically:
+for the <<mapper-orm,Hibernate ORM integration>> this happens on transaction commit,
+while for the <<mapper-pojo-standalone,Standalone POJO Mapper>> this happens when the `SearchSession` is closed.
+
+The `SearchIndexingPlan` interface offers the following methods:
+
+`add(Object entity)`::
+(Available with the <<mapper-pojo-standalone,Standalone POJO Mapper>> only.)
++
+Add a document to the index if the entity type is mapped to an index (`@Indexed`).
++
+WARNING: This may create duplicates in the index if the document already exists.
+Prefer `addOrUpdate` unless you are really sure of yourself and need a (slight) performance boost.
+`addOrUpdate(Object entity)`::
+Add or update a document in the index if the entity type is mapped to an index (`@Indexed`),
+and re-index documents that embed this entity (through `@IndexedEmbedded` for example).
+`delete(Object entity)`::
+Delete a document from the index if the entity type is mapped to an index (`@Indexed`),
+and re-index documents that embed this entity (through `@IndexedEmbedded` for example).
+`purge(Class<?> entityType, Object id)`::
+Delete the entity from the index,
+but do not try to re-index documents that embed this entity.
++
+Compared to `delete`, this is mainly useful if the entity has already been deleted from the database
+and is not available, even in a detached state, in the session.
+In that case, reindexing associated entities will be the user's responsibility,
+since Hibernate Search cannot know which entities are associated to an entity that no longer exists.
+`purge(String entityName, Object id)`::
+Same as `purge(Class<?> entityType, Object id)`,
+but the entity type is referenced by its name (see `@javax.persistence.Entity#name`).
+`process()` and `execute()`::
```

Comment:
Maybe it's better to split these into two items? since execute would write to an index in `no-coordination` case and these updates won't be rolled back in case of a transaction rollback...